zebrad/src/commands/start/sync.rs

```diff
@@ -0,0 +1,204 @@
+use color_eyre::Report;
+use eyre::eyre;
+use futures::stream::{FuturesUnordered, StreamExt};
+use std::{collections::HashSet, iter};
+use tower::{Service, ServiceExt};
+use tracing_futures::Instrument;
+use zebra_chain::{block::BlockHeaderHash, types::BlockHeight};
+
+pub struct Syncer<ZN, ZS>
+where
+    ZN: Service<zebra_network::Request>,
+{
+    pub peer_set: ZN,
+    pub state: ZS,
+    pub tip_requests: FuturesUnordered<ZN::Future>,
```

Comment:
To be able to keep track of the tips themselves, this has to be 
```suggestion
    pub prospective_tips: HashSet<BlockHeaderHash>,
```
instead of a fanned-out collection of futures.